hack/upgrade-test.sh

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env bash
+# Copyright 2021 VMware
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# shellcheck disable=SC2155
+readonly DIR="$(cd "$(dirname "$0")" && pwd)"
+readonly HOST_ADDR=${HOST_ADDR:-$("$DIR"/ip.py)}
+readonly REGISTRY_PORT=${REGISTRY_PORT:-5000}
+readonly REGISTRY=${REGISTRY:-"${HOST_ADDR}:${REGISTRY_PORT}"}
+
+readonly TEST_NAME="gitwriter-sc"
+readonly GIT_SERVER="git-server.default.svc.cluster.local:80"
+readonly SOURCE_REPO="hello-world"
+readonly SOURCE_BRANCH="main"
+readonly CONFIG_REPO="gitops-test"
+readonly CONFIG_BRANCH="main"
+readonly CONFIG_COMMIT_MESSAGE="Update config"
+
+main() {
+  ./hack/setup.sh cluster cartographer-latest example-dependencies
```

Comment:
```suggestion
  $DIR/setup.sh cluster cartographer-latest example-dependencies
```

or have a `cd $DIR` before everything

---

in the scripts we usually either:

- in `main`, change directory to an absolute path (usually the root of the repository, like `cd $ROOT`
- in functions, refer to files with absolute paths (like, `ytt $DIR/overlays/...` where `$DIR` is an absolute path to a directory)

such that it doesn't matter where we execute the scripts from

wdyt?